# Comparative Genomics

Ensembl Compara provides cross-species resources and analyses, at both the sequence level and the gene level. These data can be accessed in [various ways](accessing_compara.md).


## Gene analyses

All Ensembl gene sequences are compared to one another in order to produce gene trees, infer homologues and produce gene families. These processes are different for coding and non-coding genes.

* [Protein trees](protein_trees.md)
* [ncRNA trees](ncRNA_trees.md)
* [Homologue types](homologue_types.md)
* [Gene families](gene_families.md)
* [Orthologue quality scores](orthology_quality_controls.md)
* [Gene tree stable IDs](gene_tree_stable_id.md)


## Genome analyses

Whole genome alignments are produced, both pairwise alignments and multiple alignments. From these we calculate ancestral sequences, when bases were fixed, conservation scores, constrained elements and syntenies.

* [Pairwise alignments](pairwise_genome_alignments.md)
* [Multiple alignments](multiple_genome_alignment.md)
* [Ancestral sequences](ancestral_sequences.md)
* [Conservation scores and constrained elements](conservation_and_constrained.md)
* [Synteny](synteny.md)



## Species tree

The Compara pipelines use various species trees for different purposes. [Find out more](species_trees.md)
