# ncRNA trees

ncRNA trees are generated by a pipeline that uses a strategy similar to the one used for [protein trees](protein_trees.md), but adapted to the specific characteristics of ncRNAs, such as their secondary structure. This is important because ncRNA genes are well known to form secondary structures where pairs of residues are matched to form loops and other structures. Substitution models that consider pairs of sites have been proposed and implemented in several packages like PHASE or RAxML.

[diagram of tree building]

## Details on tree building

The ncRNA tree pipeline consists of the following steps:
1. Get and store ncRNA family models from [RFAM](http://rfam.xfam.org/)
2. Load and identify all the ncRNA members annotated in all the Ensembl genomes
3. Filter out extra copies in low-coverage assemblies usin our EPO multiple alignments3
4. Build secondary structure alignments using INFERNAL4 and refinement of the covariance model
5. Build ncRNA trees with RAxML5 using 16 different secondary structure models
6. In parallel with the secondary structure alignments and trees, build multiple alignments with PRANK6 with the genomic sequences of the ncRNAs. For these alignments we include the flanking region of the genes (2 times the length of the gene at each side).
7. With the genomic alignments, build a neighbour-joining (NJ) and a maximum-likelihood (ML) tree using TreeBeST7.
8. For very big families, build fast and efficient trees using FastTree8 and RAxML-Light
9. For each family, add the species tree to the set of trees already obtained and reconcile them all using TreeBeST7 obtaining one final tree for each family.

## References

[ncRNA orthologies in the vertebrate lineage. Miguel Pignatelli, Albert J. Vilella, Matthieu Muffato, Leo Gordon, Simon White, Paul Flicek, Javier Herrero. Database (Oxford) 2016 pii:bav127.](http://europepmc.org/articles/PMC4792531)
Vilella AJ, Severin J, Ureta-Vidal A, Heng L, Durbin R, Birney E. EnsemblCompara GeneTrees: Complete, duplication aware phylogenetic trees in vertebrates. Genome Res 2009 19:327
Gardner PP, Daub J, Tate J, Moore BL, Osuch IH, Griffiths-Jones S, Finn RD, Nawrocki EP, Kolbe DL, Eddy SR, Bateman A. Rfam: Wikipedia, clans and the "decimal" release. Nucleic Acid Res 2011, DatabaseIssue:D141
Paten B, Herrero J, Beal K, Fitzgerald S, Birney E. Enredo and Pecan: genome-wide mammalian consistency-based multiple alignment with paralogs. Genome Res 2008 18:1814
Nawrocki EP, Kolbe DL, Eddy SR. Infernal 1.0: inference of RNA alignments. Bioinformatics 2009 25:1335
Stamatakis A. RAxML-VI-HPC: maximum likelihood-based phylogenetic analyses with thousands of taxa and mixed models. Bioinformatics 2006 22:2688
Loytynoja A, Goldman N. webPRANK: a phylogeny-aware multiple sequence aligner with interactive alignment browser. BMC Bioinformatics 2010 11:579
Li H et al. TreeBeST. http://treesoft.sourceforge.net/treebest.shtml
Price MN, Dehal PS, Arkin AP. FastTree 2--aproximately maximum-likelihood trees for large alignments. PLoS One 2010 5:e9490
